homeassistant/components/goodwe/button.py

```diff
@@ -0,0 +1,84 @@
+"""GoodWe PV inverter selection settings entities."""
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class GoodweButtonEntityDescriptionRequired:
+    """Required attributes of GoodweButtonEntityDescription."""
+
+    action: Callable[[Inverter], Awaitable[None]]
+
+
+@dataclass
+class GoodweButtonEntityDescription(
+    ButtonEntityDescription, GoodweButtonEntityDescriptionRequired
+):
+    """Class describing Goodwe button entities."""
+
+
+SYNCHRONIZE_CLOCK = GoodweButtonEntityDescription(
+    key="synchronize_clock",
+    name="Synchronize inverter clock",
+    icon="mdi:clock-check-outline",
+    entity_category=EntityCategory.CONFIG,
+    action=lambda inv: inv.write_setting("time", datetime.now()),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the inverter button entities from a config entry."""
+    inverter = hass.data[DOMAIN][config_entry.entry_id][KEY_INVERTER]
+    device_info = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE_INFO]
+
+    # read current time from the inverter
+    try:
+        await inverter.read_setting("time")
+    except (InverterError, ValueError):
+        # Inverter model does not support clock synchronization
+        _LOGGER.debug("Could not read inverter current clock time")
```

Comment:
Please raise some error here, so the system knows the entry setup failed.

Poossibly the not ready exception to try again later.

Reply:
This is intentional, not all inverter models support this feature, in such case, the button is just not created at all.
This is not exceptional state, this is expected state (in some inverters).